qutebrowser/browser/downloads.py

```diff
@@ -507,6 +507,25 @@ def __str__(self):
                         remaining=remaining, perc=perc, down=down,
                         total=total, errmsg=errmsg))
 
+    def _find_next_filename(self) -> None:
+        """Append unique numeric suffix to filename.
+
+        After finding a unique filename, restart the process of setting the
+        filename which will update the filename shown in the downloads list."""
+        assert self._filename is not None
+        path = pathlib.Path(self._filename)
+        suffix = ''.join(path.suffixes)
+        base = path.name[:-len(suffix)]
+
+        i = 1
+        while True:
+            i += 1
```

Comment:
Please limit this using

```suggestion
        for i in range(100):
```

or so, and then use a [for-else](https://book.pythontips.com/en/latest/for_-_else.html) to figure out some error handling (probably call `self._die`?) if we didn't find a matching filename.

Yes, this usually shouldn't happen - but filesystems (with FUSE and such) can act in unexpected ways, and having an endless loop here when a filesystem claims all those variants exist would be bad.